roles/internetarchive/tasks/main.yml

```diff
@@ -1,32 +1,93 @@
-# We need a recent version of node
-- name: FAIL (STOP INSTALLING) IF nodejs_version is not set to 10.x or 12.x
-  fail:
-    msg: "Internet Archive install cannot proceed, as it currently requires Node.js 10.x or 12.x, and your nodejs_version is set to {{ nodejs_version }}.  Please check the value of nodejs_version in /opt/iiab/iiab/vars/default_vars.yml and possibly also /etc/iiab/local_vars.yml"
-  when: internetarchive_install and (nodejs_version != "10.x") and (nodejs_version != "12.x")
+# "How do i fail a task in Ansible if the variable contains a boolean value?
+# I want to perform input validation for Ansible playbooks"
+# https://stackoverflow.com/questions/46664127/how-do-i-fail-a-task-in-ansible-if-the-variable-contains-a-boolean-value-i-want/46667499#46667499
 
-- name: Install Yarn and Internet Archive
+# We assume 0-init/tasks/validate_vars.yml has DEFINITELY been run, so no need
+# to re-check whether vars are defined here.  As Ansible vars cannot be unset:
+# https://serverfault.com/questions/856729/how-to-destroy-delete-unset-a-variable-value-in-ansible
+
+- name: Assert that "internetarchive_install is sameas true" (boolean not string etc)
+  assert:
+    that: internetarchive_install is sameas true
+    fail_msg: "PLEASE SET 'internetarchive_install: True' e.g. IN: /etc/iiab/local_vars.yml"
+    quiet: yes
+
+- name: Assert that "internetarchive_enabled | type_debug == 'bool'" (boolean not string etc)
+  assert:
+    that: internetarchive_enabled | type_debug == 'bool'
+    fail_msg: "PLEASE GIVE VARIABLE 'internetarchive_enabled' A PROPER (UNQUOTED) ANSIBLE BOOLEAN VALUE e.g. IN: /etc/iiab/local_vars.yml"
+    quiet: yes
+
+
+#- name: Set --reinstall fact
+#  set_fact:
+#    internetarchive_upgrade: True
+#  when: reinstall is defined
+
+- block:    # UPDATE IF... internetarchive_installed is defined and internet_available
+
+  - name: Stop 'internetarchive' systemd service, if internetarchive_upgrade
+    systemd:
+      name: internetarchive
+      daemon_reload: yes
+      state: stopped
+    #when: internetarchive_enabled and internetarchive_upgrade
+
```

Comment:
This change should cause the install to fail as the service file is not installed yet, that is done later in install.yml but ansible might silently discard the `Failed to stop foo.service: Unit foo.service not loaded` error emitted by systemd, I think that would be a bug in ansbile.

Reply:
Install doesn't fail fyi, as confirmed in my testing a fresh install.

See the enclosing `block:` (indented part) which prevents those 2 stanzas from running if internetarchive_installed is not yet defined.

Reply:
I see, something new one big when clause in place of using two whens, missed that.